### PR TITLE
Remove hardcoded github provider

### LINKED ISSUE #34: GitHub provider not supported on arm64 macs (M1)
```
│ Error: Incompatible provider version
│ 
│ Provider registry.terraform.io/hashicorp/github v3.0.0 does not have a
│ package available for your current platform, darwin_arm64.
│ 
│ Provider releases are separate from Terraform CLI releases, so not all
│ providers are available for all platforms. Other versions of this provider
│ may have different platforms supported.
```

### PR DESCRIPTION
**Pull request by drselump14**
- Remove detailed GitHub provider
- Allow GitHub provider to be injected

Resolves #34 
Follows up for PR #35

### CODE REVIEW

**Comment by nitrocode** [comment]
@drselump14 please add the repo root's versions.tf to examples/complete directory to fix test/bats and get the test/terratest to run.

File: examples/complete/versions.tf [hunk 1]
```diff
@@ -0,0 +1 @@
+versions.tf
```
Comment (nitrocode) [inline examples/complete/versions.tf]: This is incorrect. Please copy and paste the versions.tf in the repo root to this directory.

Comment (drselump14) [reply examples/complete/versions.tf]: My bad, let me fix it

Comment (nitrocode) [reply examples/complete/versions.tf]: No worries, I fixed it.

File: examples/complete/versions.tf [hunk 2]
```diff
@@ -0,0 +1,16 @@
+terraform {
```
Comment (nitrocode) [inline examples/complete/versions.tf:1]: I'm unsure why this file cannot be read by the auto format bot.

Could you run `make pr/auto-format` locally?

File: versions.tf [hunk 3]
```diff
@@ -3,10 +3,10 @@ terraform {
 
   required_providers {
     github = {
-      source = "hashicorp/github"
+      source = "integrations/github"
       # breaking changes both immediately before and after 3.0.0, pin exactly
       # until ready to upgrade to integrations/github 4.2 or later
-      version = "3.0.0"
```
Comment (nitrocode) [inline versions.tf:9]: We can remove the 2 comments above

File: examples/complete/versions.tf (hunk 2 above)
Comment (drselump14) [reply examples/complete/versions.tf:1]: ok, will do

File: versions.tf (hunk 3 above)
Comment (drselump14) [reply versions.tf:9]: Done

File: examples/complete/versions.tf (hunk 2 above)
Comment (drselump14) [reply examples/complete/versions.tf:1]: I run `make pr/auto-format` successfully on my local environment

File: examples/complete/main.tf [hunk 4]
```diff
@@ -2,14 +2,21 @@ provider "aws" {
...
-  github_organization = var.github_organization
   github_repositories = var.github_repositories
   webhook_url         = var.webhook_url
   events              = var.events
 
   context = module.this.context
+
+  providers = {
+    github = github
+  }
```
Comment (nitrocode) [inline examples/complete/main.tf]: This is not needed since there isn't an alias, the provider should be implicitly provided

**Comment by nitrocode** [comment]
There is a [terratest error](https://github.com/cloudposse/actions/actions/runs/3511814454). Looks like it doesn't show an org in the api call which is odd since you're correctly passing in the `owner` and the `integrations/github` provider states in their docs that the [organization](https://registry.terraform.io/providers/integrations/github/latest/docs#organization) input is deprecated.

```
Error: POST https://api.github.com/repos//terraform-github-repository-webhooks/hooks: 404 Not Found []
```

I'm unsure why the org is missing here.

Please troubleshoot and run the tests locally to reproduce and resolve.

**Comment by drselump14** [comment]
> There is a [terratest error](https://github.com/cloudposse/actions/actions/runs/3511814454). Looks like it doesn't show an org in the api call which is odd since you're correctly passing in the `owner` and the `integrations/github` provider states in their docs that the [organization](https://registry.terraform.io/providers/integrations/github/latest/docs#organization) input is deprecated.
> 
> ```
> Error: POST https://api.github.com/repos//terraform-github-repository-webhooks/hooks: 404 Not Found []
> ```
> 
> I'm unsure why the org is missing here.
> 
> Please troubleshoot and run the tests locally to reproduce and resolve.

I'm unsure how to run terratest locally, but I can confirm that `terraform apply --var-file fixtures.us-east-2.tfvars` works with the latest commit.

**Comment by nitrocode** [comment]
cc: @Nuru please review when you get a chance. Requesting your help since you gave an in-depth set of steps in the previous PR and @drselump14 has taken it on to complete each one. 

The tests pass and so far this looks pretty good to me. I think a second set of eyes would help here.

**Review by nitrocode — APPROVED** [review]
Nice job!

**Comment by mihaiplesa** [comment]
Looking forward to this.

**Comment by adamantike** [comment]
Can this change be considered for merge and release? It would also fix #37, which is avoiding higher-level modules (like [cloudposse/terraform-aws-ecs-web-app](https://github.com/cloudposse/terraform-aws-ecs-web-app)) from using `count` or `for_each`.

**Pull request merged**